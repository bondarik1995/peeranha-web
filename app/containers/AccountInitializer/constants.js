/*
 *
 * AccountInitializer constants
 *
 */

export const GET_CURRENT_ACCOUNT = 'app/AccountInitializer/GET_CURRENT_ACCOUNT';
export const GET_CURRENT_ACCOUNT_SUCCESS =
  'app/AccountInitializer/GET_CURRENT_ACCOUNT_SUCCESS';
export const GET_CURRENT_ACCOUNT_ERROR =
  'app/AccountInitializer/GET_CURRENT_ACCOUNT_ERROR';
