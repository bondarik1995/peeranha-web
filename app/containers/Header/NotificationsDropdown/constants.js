export const ROW_HEIGHT = 70;
export const MENU_WIDTH = 330;
export const MENU_HEIGHT = 280;
export const HEADER_AND_FOOTER_HEIGHT = 104;
export const THRESHOLD = 0.4;
