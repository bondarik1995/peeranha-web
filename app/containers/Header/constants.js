import { isSingleCommunityWebsite } from 'utils/communityManagement';

const singleCommunityId = isSingleCommunityWebsite();

export const HEADER_ID = 'containers_header_HEADER_ID';
export const HEADER_HEIGHT = singleCommunityId ? 120 : 80;
export const MOBILE_HEADER_HEIGHT = singleCommunityId ? 90 : 60;
