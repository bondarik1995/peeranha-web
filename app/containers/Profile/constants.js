/*
 *
 * Profile constants
 *
 */

export const GET_PROFILE_INFORMATION = 'app/Profile/GET_PROFILE_INFORMATION';
export const GET_PROFILE_INFORMATION_SUCCESS =
  'app/Profile/GET_PROFILE_INFORMATION_SUCCESS';
export const GET_PROFILE_INFORMATION_ERROR =
  'app/Profile/GET_PROFILE_INFORMATION_ERROR';

export const UPLOAD_IMAGE_FILE = 'app/Profile/UPLOAD_IMAGE_FILE';
export const UPLOAD_IMAGE_FILE_SUCCESS =
  'app/Profile/UPLOAD_IMAGE_FILE_SUCCESS';
export const UPLOAD_IMAGE_FILE_ERROR = 'app/Profile/UPLOAD_IMAGE_FILE_ERROR';
