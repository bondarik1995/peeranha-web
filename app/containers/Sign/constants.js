/*
 *
 * Sign constants
 *
 */

export const DEFAULT_ACTION = 'app/Sign/DEFAULT_ACTION';
export const EOS_ACC = 'app/Sign/eosAccount';
export const DISPLAY_NAME = 'app/displayName';
