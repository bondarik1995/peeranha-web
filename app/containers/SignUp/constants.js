/*
 *
 * SignUp constants
 *
 */

export const DEFAULT_ACTION = 'app/SignUp/DEFAULT_ACTION';
export const EOS_ACC = 'app/SignUp/eosAccount';
export const DISPLAY_NAME = 'app/SignUp/displayName';

export const FETCH_REGISTR_ACC = 'app/SignUp/FETCH_REGISTR_ACC';
export const REGISTR_ACC_SUCCESS = 'app/SignUp/REGISTR_ACC_SUCCESS';
export const REGISTR_ACC_ERROR = 'app/SignUp/REGISTR_ACC_ERROR';
export const SET_REDUCER_DEFAULT = 'app/SignUp/SET_REDUCER_DEFAULT';
