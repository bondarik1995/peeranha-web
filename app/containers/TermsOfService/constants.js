export const GET_TERMS = 'app/PrivacyPolicy/GET_TERMS';
export const GET_TERMS_SUCCESS = 'app/PrivacyPolicy/GET_TERMS_SUCCESS';
export const GET_TERMS_ERROR = 'app/PrivacyPolicy/GET_TERMS_ERROR';

export const SECTION_ID = 'terms-of-service';
