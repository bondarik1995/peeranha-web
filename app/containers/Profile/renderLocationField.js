import React from 'react';
import PropTypes from 'prop-types';

import LocationList from './LocationList';
import { LOCATION_FIELD } from './constants';

const cities = (list, sendProps) =>
  list.map(item => (
    <li
      role="presentation"
      key={item.geonameId}
      onClick={() =>
        sendProps.chooseLocation(
          item.geonameId,
          `${item.name}, ${item.countryName}`,
        )
      }
    >{`${item.name}, ${item.countryName}`}</li>
  ));

/* eslint-disable */
function renderLocationField({ input, label, sendProps }) {
  const { ipfs } = sendProps.profile;

  return (
    <div>
      <h6>{label}</h6>
      <div className="position-relative mb-0">
        <input
          {...input}
          readOnly={!sendProps.isOwner}
          autoComplete="off"
          placeholder={label}
          onChange={e => sendProps.getCitiesList(e.target.value)}
          value={ipfs[LOCATION_FIELD] && ipfs[LOCATION_FIELD].name}
          className="form-control mb-0"
          type="text"
        />
        {ipfs[LOCATION_FIELD] &&
          !ipfs[LOCATION_FIELD].id &&
          ipfs[LOCATION_FIELD].name && (
          <LocationList>
            {cities(sendProps.citiesList, sendProps)}
          </LocationList>
        )}
      </div>
    </div>
  );
}
/* eslint-enable */

renderLocationField.propTypes = {
  input: PropTypes.object.isRequired,
  label: PropTypes.string.isRequired,
  sendProps: PropTypes.object.isRequired,
};

export default renderLocationField;
