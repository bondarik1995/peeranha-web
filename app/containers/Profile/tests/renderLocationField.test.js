import renderer from 'react-test-renderer';
import 'jest-styled-components';

import renderLocationField from '../renderLocationField';
import { LOCATION_FIELD } from '../constants';

describe('renderLocationField test', () => {
  it('test by snapshots', () => {
    const renderedComponent = renderer
      .create(
        renderLocationField({
          input: {},
          label: 'string',
          sendProps: {
            profile: {
              ipfs: {
                [LOCATION_FIELD]: {
                  name: '',
                },
              },
            },
          },
        }),
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
