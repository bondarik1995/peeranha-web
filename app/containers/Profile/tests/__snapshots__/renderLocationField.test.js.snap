// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderLocationField test test by snapshots 1`] = `
<div>
  <h6>
    string
  </h6>
  <div
    className="position-relative mb-0"
  >
    <input
      autoComplete="off"
      className="form-control mb-0"
      onChange={[Function]}
      placeholder="string"
      readOnly={true}
      type="text"
      value=""
    />
    
  </div>
</div>
`;
